core/src/main/scala/com/avast/grpc/jsonbridge/ReflectionGrpcJsonBridge.scala

```diff
@@ -2,93 +2,105 @@ package com.avast.grpc.jsonbridge
 
 import java.lang.reflect.Method
 
-import cats.effect.Async
-import cats.syntax.all._
+import cats.effect._
+import cats.implicits._
 import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcMethodName
-import com.google.common.util.concurrent._
+import com.google.common.util.concurrent.{FutureCallback, Futures, ListenableFuture}
 import com.google.protobuf.util.JsonFormat
 import com.google.protobuf.{Message, MessageOrBuilder}
 import com.typesafe.scalalogging.StrictLogging
 import io.grpc.MethodDescriptor.{MethodType, PrototypeMarshaller}
-import io.grpc._
 import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc._
 import io.grpc.stub.AbstractStub
 
 import scala.collection.JavaConverters._
 import scala.concurrent.ExecutionContext
-import scala.language.{existentials, higherKinds}
+import scala.language.existentials
+import scala.language.higherKinds
 import scala.util.control.NonFatal
 
-class ReflectionGrpcJsonBridge[F[_]](services: ServerServiceDefinition*)(implicit ec: ExecutionContext, F: Async[F])
-    extends GrpcJsonBridge[F]
-    with AutoCloseable
-    with StrictLogging {
-
-  import com.avast.grpc.jsonbridge.ReflectionGrpcJsonBridge._
+object ReflectionGrpcJsonBridge extends StrictLogging {
 
-  def this(grpcServer: io.grpc.Server)(implicit ec: ExecutionContext, F: Async[F]) = this(grpcServer.getImmutableServices.asScala: _*)
+  // JSON body and headers to a response (fail status or JSON response)
+  private type HandlerFunc[F[_]] = (String, Map[String, String]) => F[Either[Status, String]]
```

Comment:
I would make this type public because you use it in one public method parameters (I'm even surprised it's not required because of that, probably because it's _just a type alias_).

Reply:
I was initially considering removing the type alias, but then I kept it (private), because it shortens type annotations in our implementation
but I don't think we should bother outside world with it -- `(String, Map[String, String]) => F[Either[Status, String]]` is quite straightforward and self-explanatory

Reply:
But you used this alias in on of the public methods ;-)

Reply:
yes, but the method is perfectly usable ;)